server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -3816,14 +3816,6 @@ public PhysicalNetwork createPhysicalNetwork(final Long zoneId, final String vne
             throw new InvalidParameterException("Only one isolationMethod can be specified for a physical network at this time");
         }
 
-        if (vnetRange != null) {
-            // Verify zone type
-            if (zoneType == NetworkType.Basic || (zoneType == NetworkType.Advanced && zone.isSecurityGroupEnabled())) {
```

Comment:
@weizhouapache we removing Basic zone check as well. Is this okay?

Reply:
Good point, thanks @shwstppr 
I will create a Basic zone and test it.

Reply:
added back the check for Basic zone

I was not able to create L2 networks in Basic zone, it is unnecessary to set vlan range in Basic zone
```
(localcloud) 🐱 > create network name=L2 displaytext=L2 zoneid=10ac2ce3-05b9-4cd3-b920-73de0c47b4bf networkofferingid=3d7d17c6-24c4-4ead-8490-bec6bd6d0779
🙈 Error: (HTTP 431, error code 4350) Only AclType=Domain can be specified for network creation in Basic zone
```